### DISCUSSION TITLE
Sql Joins

### DISCUSSION THREAD

**Discussion by KellerKev** [emmett-framework/emmett#336] [Q&A] [answered]
In my test project I am coming across the need to do a left outer join.

Could you help me how I can do it with Emmett?

Here is an example from the pydal world

rows = db().select(db.person.ALL, db.thing.ALL,
...                    left=db.thing.on(db.person.id == db.thing.owner_id))
>>> for row in rows:
...     print row.person.name, 'has', row.thing.name
... 
Alex has Boat
Alex has Chair
Bob has Shoes
Carl has None

On a side note maybe we can also talk about how we can do an inner join while I have you and finally last question how we can use the query values of the join operation in the template. 

In the template I need to know which table I am using.
If I put the join query in a variable rows and iterate in the template I would usually end up like this

Row.person.name
Row.thing.name

To render the name of the thing that belongs to the person with a name X. 

Finally here is the example of an inner join I was wondering how you souls do it in Emmett. 

rows = db(db.person.id == db.thing.owner_id).select()
>>> for row in rows:
...     print row.person.name, 'has', row.thing.name
... 
Alex has Boat
Alex has Chair
Bob has Shoes

**Comment by KellerKev** [comment]
From the documentation I suppose has_many could fit the bill.

**Comment by gi0baro** [comment, marked as answer]
@Kkeller83 you can find the methods in docs @ https://emmett.sh/docs/2.2.x/orm/relations#joins-and-n1-queries

So in case you use the ORM relations, you have the `join()` method for inner joins and the `select(including='related')` for LEFT OUTER joins. In case you don't specify relations in your models, the pyDAL syntax should still be valid.

**Comment by KellerKev** [comment]
Excellent, thanks!
